### PR TITLE
core_rpc_server: return ID of submitted block

### PR DESCRIPTION
**Pull request by jeffro256**
Closes https://github.com/monero-project/monero/issues/8874

### CODE REVIEW

**Review by selsta — COMMENTED** [review]
Somehow this breaks the mining test


```
simplejson.errors.JSONDecodeError: Invalid control character '\x06' at: line 5 column 18 (char 70)
```

**Comment by jeffro256** [comment]
Sorry I forgot `KV_SERIALIZE_VAL_POD_AS_BLOB` doesn't parse to/from hex. Will fix in a couple minutes

**Comment by jeffro256** [comment]
Made the field hex and added functional tests

File: src/rpc/core_rpc_server_commands_defs.h [hunk 1]
```diff
@@ -1115,8 +1115,11 @@ namespace cryptonote
     
     struct response_t: public rpc_response_base
     {
+      std::string block_id;
+
       BEGIN_KV_SERIALIZE_MAP()
         KV_SERIALIZE_PARENT(rpc_response_base)
+        KV_SERIALIZE(block_id)
```
Comment (SChernykh) [inline src/rpc/core_rpc_server_commands_defs.h:1122]: You should increase `CORE_RPC_VERSION_MINOR` if you change RPC definitions.

**Review by SChernykh — CHANGES_REQUESTED** [review]

File: src/rpc/core_rpc_server_commands_defs.h (hunk 1 above)
Comment (jeffro256) [reply src/rpc/core_rpc_server_commands_defs.h:1122]: Good point, updated

**Review by SChernykh — APPROVED** [review]

**Pull request merged**